apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/AbstractTracingSpan.java

```diff
@@ -322,4 +320,19 @@ public AbstractSpan start(long startTime) {
             refs.add(ref);
         }
     }
+
+    @Override public AbstractSpan prepareForAsync() {
+        context = ContextManager.awaitFinishAsync(this);
```

Comment:
`isInAsyncMode` is never set to true in this class. Did you mean:
```suggestion
        isInAsyncMode = true;
        context = ContextManager.awaitFinishAsync(this);
```